cilium-health/launch/launcher.go

```diff
@@ -60,44 +89,59 @@ func (ch *CiliumHealth) Run() {
 		time.Sleep(connectRetryInterval)
 	}
 
+	// When the unix socket is made available, set its permissions.
+	os.Remove(defaults.SockPath)
+	go func() {
+		scopedLog := log.WithField(logfields.Path, defaults.SockPath)
+		for {
+			_, err := os.Stat(defaults.SockPath)
+			if err == nil {
+				break
+			}
+			scopedLog.WithError(err).Debugf("Cannot find socket")
+			time.Sleep(1 * time.Second)
+		}
+		if err := api.SetDefaultPermissions(defaults.SockPath); err != nil {
+			scopedLog.WithError(err).Fatal("Cannot set default permissions on socket")
+		}
+	}()
+
+	// Launch cilium-health API server
+	go func() {
+		defer ch.server.Shutdown()
+		if err := ch.server.Serve(); err != nil {
+			log.WithError(err).Error("Failed to serve cilium-health API")
+		}
+	}()
+
+	// Periodically fetch status from cilium-health server
 	for {
-		var err error
+		status := &models.Status{
+			State: models.StatusStateOk,
+		}
 
-		os.Remove(defaults.SockPath)
-		ch.Launcher.Run()
-		ch.client, err = healthPkg.NewDefaultClient()
+		_, err := ch.client.Restapi.GetHealthz(nil)
 		if err != nil {
-			log.WithError(err).Infof("Cannot establish connection to local %s instance", targetName)
-			ch.Launcher.Stop()
-			time.Sleep(connectRetryInterval)
-			continue
+			status.Msg = ciliumPkg.Hint(err).Error()
```

Comment:
Do we need the `Hint()` here? This is now run as part of the daemon, so the daemon must be running.